source/uwp/Renderer/lib/AdaptiveInputs.cpp

```diff
@@ -85,12 +88,16 @@ namespace AdaptiveNamespace
     {
         ComPtr<IAdaptiveInputElement> localInputElement(inputElement);
 
-        ComPtr<AdaptiveInputElementBase> inputElementImpl;
-        RETURN_IF_FAILED(localInputElement.As(&inputElementImpl));
+        ComPtr<IAdaptiveCardElement> cardElement;
+        RETURN_IF_FAILED(localInputElement.As(&cardElement));
+
+        HString elementId;
+        RETURN_IF_FAILED(cardElement->get_Id(elementId.GetAddressOf()));
 
-        InternalId inputId = inputElementImpl->GetInternalId();
+        std::string utf8Id;
+        RETURN_IF_FAILED(HStringToUTF8(elementId.Get(), utf8Id));
```

Comment:
Can you explain this change a bit? So we are not working with the internalId but card id now?

Reply:
custom elements don't have an internal id (or they may but it's not accessible), that breaks the first solution I came up with, now we'll use the input Id for registering the input, this will increase the search complexity but at least we know that all inputs have an unique Id is an enforced requirement